src/main.py

```diff
@@ -0,0 +1,201 @@
+import yaml
+from telebot import TeleBot
+from telebot.types import BotCommand, Message
+from data_collector.camera_utils import get_camera, get_camera_image
+from data_collector.gdrive import upload_photo
+from enum import StrEnum
+import time
+from io import BytesIO
+from PIL.Image import Image
+import threading
+
+
+class CaptureStatus(StrEnum):
+    NOT_STARTED = "not started"
+    RUNNING = "running"
+
+
+current_status: CaptureStatus = CaptureStatus.NOT_STARTED
+capture_thread: threading.Thread | None = None
+capture_thread_duration: int = -1
+
+
+def _peek(message: Message, bot: TeleBot) -> None:
+    """
+    This function sends the current camera view
+    :param message: The message object
+    :param bot: The bot object
+    """
+    peek: Image | None = get_camera_image()
+    if peek is not None:
+        image_bytes = BytesIO()
+        image_bytes.name = 'peek.jpeg'
+        peek.save(image_bytes, 'JPEG')
+        image_bytes.seek(0)
+        bot.send_photo(message.chat.id, photo=image_bytes, message_thread_id=message.message_thread_id)
+    else:
+        bot.send_message(message.chat.id, "Unable to capture image",
+                         message_thread_id=message.message_thread_id)
+
+
+def capture_job(duration: int) -> None:
+    """
+    This function captures images from the camera every 30 seconds. The process will run for the specified duration.
+    If duration is 0, the process will run indefinitely
+    """
+    global current_status, capture_thread_duration
+    internal_counter: int = 0
+    endless = duration == 0
+    duration = duration * 60
```

Comment:
`duration *= 60`
in termini di memoria risparmi ([source](https://stackoverflow.com/questions/41446833/what-is-the-difference-between-i-i-1-and-i-1-in-a-for-loop))